File: chapter05/akkademaid-java-5/src/main/java/com/lp/akka/notes/ArticleParseActor.java
package com.lp.akka.notes;

import akka.actor.AbstractActor;
import akka.actor.Status;
import akka.japi.pf.ReceiveBuilder;

/**
 * @author li.pan
 * @version 1.0.0
 * @Description TODO
 * @createTime 2021年01月14日 12:59:00
 */
public class ArticleParseActor extends AbstractActor {

    private ArticleParseActor() {
        receive(ReceiveBuilder.
                match(ParseArticle.class, x ->{
                            ArticleParser.apply(x.htmlBody).
                                    onSuccess(body -> sender().tell(body, self())).
                                    onFailure(t -> sender().tell(new Status.Failure(t), self()));
                        }
                ).
                build());
    }
}

File: chapter05/akkademaid-scala-5/src/main/scala/com/lp/akka/notes/ArticleParseActor.scala
package com.lp.akka.notes

import akka.actor.Actor

/**
 * @author li.pan
 * @version 1.0.0
 * @Description TODO
 * @createTime 2021年01月14日 12:56:00
 */
class ArticleParseActor extends Actor {
  override def receive: Receive = {
    case ParseArticle(htmlString) =>
      val body: String = ArticleParser(htmlString)
      sender() ! body
  }
}

object ArticleParser {
  def apply(html: String): String =
    de.l3s.boilerpipe.extractors.ArticleExtractor.INSTANCE.getText(html)
}

case class ParseArticle(htmlBody: String)